functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,276 @@ export const processNewAllotment = functions.database
     });
     return 1;
   });
+
+
+
+const validateSpreadsheetRow = (row, summary, sheetTitle) => {
+    let resolutions = ['ok', 'drop', 'duplicate', 'on hold', 'reallot', 'repeat', 'derivative'];
+    let resolution, fidCheckRes; // variables to hold values of Res and fidRes when looping through the rows
+
+    let _import = false;
+ 
+
+    // [CHECK #1] File name belongs to the list identified by the sheet name       
+    if (helpers.extractListFromFilename(row['Audio File Name']) != sheetTitle) {
+        console.log("File name belongs to a different list.");
+        summary.misplaced.push({
+            audioFileName: row['Audio File Name'],
+            listName: sheetTitle,
+            errorMessage: `File name belongs to a different list.`
+        });
+    }
+
+    resolution = row['Resolution'];
+    fidCheckRes = row['Fidelity Check Resolution'];
+
+    if (resolution && fidCheckRes) {
+        resolution = resolution.toLowerCase();
+        fidCheckRes = fidCheckRes.toLowerCase();
+
+        // [CHECK #2] Both resolutions belong to the list of resolutions, case-insensitively.
+        if (resolutions.indexOf(resolution) < 0 || resolutions.indexOf(fidCheckRes) < 0) {
+            console.log(`Audio file name: ${row['Audio File Name']} -- Error: Resolution, or Fidelity check resolution values isn't valid`);
+            summary.invalidResoultion.push({
+                audioFileName: row['Audio File Name'],
+                resolution,
+                fidCheckRes,
+                errorMessage: `Resolution, or Fidelity check resolution values isn't valid.`
+            });
+        }
+
+        // [CHECK #3] Resolution and Fidelity Check Resolution should be equal, case-insensitively.
+        if (resolution !== fidCheckRes) {
+            console.log(`Audio file name: ${row['Audio File Name']} -- Error: Mismatch in Resolution and Fidelity check resolution values.`);
+            summary.mismatchedResolutions.push({
+                audioFileName: row['Audio File Name'],
+                resolution,
+                fidCheckRes,
+                errorMessage: `Mismatch in Resolution and Fidelity check resolution values.`
+            });
+        }
+
+        // [CHECK #6] Beginning and Ending should be filled for rows with Resolution equal to `OK`.
+        if (resolution === 'ok') {
+            _import = true;
+        }
+
+    } else // [Filter criterion #1] Rows not having (Resolution, or Fidelity check resolutin) are skipped SILENTLY
+        summary.noResolution.push({ audioFileName: row['Audio File Name'], resolution, fidCheckRes });
```

Comment:
No need reporting this case, as there will be hundreds and thousands of such files.